api/v1alpha1/openapi.yaml

```diff
@@ -2784,95 +2749,68 @@ components:
         - "FileOperationRemove" # by convention remove is fired before the file is removed.
         - "FileOperationUpdate"
         - "FileOperationReboot"
+    HookConditionalAction:
+      allOf:
+        - $ref: '#/components/schemas/HookCondition'
+        - $ref: '#/components/schemas/HookAction'
     HookAction:
       type: object
-      oneOf:
-        - properties:
-            executable:
-              $ref: '#/components/schemas/HookActionExecutableSpec'
-          required:
-           - "executable"
-        - properties:
-            systemd:
-              $ref: '#/components/schemas/HookActionSystemdSpec'
-          required:
-           - "systemd"
-    HookActionSpec:
-      type: object
-      properties:
-        timeout:
-             type: string
-             pattern: '^[1-9]\d*[smhd]$'
-             description: |
-              The maximum duration allowed for the action to complete. 
-              The duration should be specified as a positive integer 
-              followed by a time unit. Supported time units are:
-              - 's' for seconds
-              - 'm' for minutes
-              - 'h' for hours
-              - 'd' for days
-    HookActionSystemdSpec:
-       allOf:
-        - $ref: '#/components/schemas/HookActionSpec'
+      allOf:
+        - oneOf:
+            - $ref: '#/components/schemas/HookActionRun'
+            # extend hook actions
         - type: object
           properties:
-            unit:
-              $ref: '#/components/schemas/HookActionSystemdUnit'
-          required:
-            - unit
-    HookActionSystemdUnit:
-      type: object
-      properties:
-        name:
-          type: string
-          description: 'The name of the systemd unit on which the specified operations will be performed. This should be the exact name of the unit file, such as example.service. If the name is not populated the name will be auto discovered from the file path.'
-        operations:
+            timeout:
+              type: string
+              pattern: '^[1-9]\d*[smhd]$'
+              description: |
+                The maximum duration allowed for the action to complete. 
+                The duration should be specified as a positive integer 
+                followed by a time unit. Supported time units are:
+                - 's' for seconds
+                - 'm' for minutes
+                - 'h' for hours
+                - 'd' for days
```

Comment:
We should consider dropping days as they are not consistent measurements of time.

https://github.com/golang/go/issues/17767#issuecomment-258167205

Reply:
If we are not supporting days, this will impact the CSR CLI where we decided to support them to the exclusion of other units.

Reply:
Can we define a day to be 24 hours?

Reply:
I think that is fine but what is the use case for a timeout that would be over 24 hours? Even in a disconnected environment getting a signal that there is a problem and then retrying feels a bit more reasonable than a timeout lasting multiple days.

Reply:
> If we are not supporting days, this will impact the CSR CLI where we decided to support them to the exclusion of other units.

I don't see any relationship between the two honestly.